doc/network.tex

```diff
@@ -135,118 +138,132 @@
+A similar, but more informal property applies to performance of the Ouroboros network as well.
 
-A peer to peer network, is a system which provides a service to its participants while at the same
-time the participants of the network provide a service to the network.
-The performance of the P2P network depends (among other things) on the performance of the nodes,
+The network provides a service to its participants while at the same
+time the participants provide a service to the network.
+The performance of the Ouroboros network depends (among other things) on the performance of the nodes,
 while the performance of a node also depends on the performance of the network.
 Not only are there honest and adversarial participant, but there is also a huge variety of
 possible network topographies, bandwidths and latencies of network connections and other
 factors that determine the performance of the network.
-This document discusses the high level business requirements for Cardano and what are the
-assumptions on the structure of the underlying P2P network.
+
+This document discusses the high level functional and performance requirements for Ourobors and the
+assumptions made about the structure of the underlying P2P network.
 
 \subparagraph{Protocol vs Implementation}
-Network Protocols are written on different levels of abstractions.
-To be useful a protocol must be precise enough to be implemented.
-At the same time a protocol should not only be precise but also abstract.
-A high level of abstraction allows different alternative implementations of the protocol
-and also makes it possible to develop and improve it.
-Furthermore, it should not only, in principle, be possible to implement an
+Network Protocols are written at different levels of abstraction.
+To be useful, a protocol description must be precise enough to be implemented.
+A protocol description should also be abstract enough to allow alternative implementations of the protocol
+and to facilitate developing and improving it.
+Furthermore, it should be possible to both implement an
 abstract version of a protocol
-but also faithfully interpret it in a real world scenario.
+and interpret it in a real-world scenario.
 For example, it must be possible to implement a protocol
-such that the real world software runs on a real world machine with a common size of main memory
-and a common speed network adaptor.
-The Shelley network protocol has been developed in parallel with a reference implementation in Haskell.
+such that the real-world software runs on a machine with a typical size of memory
+and a typical speed network connection.
+
+The Shelley network protocol design has been developed in parallel
+with a reference implementation in Haskell.
 Haskell (more precisely GHC) has built-in support for high level concurrency abstractions
-like, for example, light-weight threads and software transactional memory.
+such as light-weight threads and software transactional memory.
```

Comment:
threads and support for compositional and concurrency safe access to shared state through software transactional memory.